### PR TITLE
Resolve push notification dependency loop

### PR DESCRIPTION
**Pull request by Julesssss**
### Details

A new developer warning message was introduced by https://github.com/Expensify/App/pull/14588. It is not a regression or bug, but should be fixed. This is due to two separate files that import one another. I simply refactored these files and function calls to prevent this. 

Warning A | Warning B 
-|-
Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-31 at 23 00 57 | Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-31 at 23 00 45 | 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/259190

### Tests

**_Testing notifications is a bit tricky, here is how I tested each supported platform:_** 
- Web:  Dev, using the `http://127.0.0.1:8080/` URL
- Desktop: Created a staging build with `npm run desktop-build-staging`, then located the app build under 'desktop-build/mac/'
- Android: [Release build,](https://stackoverflowteams.com/c/expensify/a/14785/297) against production env
- iOS: Dev, using XCode and a physical device

**Confirm users are re-registered for notifications when signing out**
_Notifications are notoriously tricky to test. I used staging for Desktop, dev for web, and physical device release builds for iOS and Android_
- Sign in as **userA** on the test platform
- As **userB**, send **userA** a message
- Notification should be show
- As **userA**, sign out
- Without clearing data or killing the window/app, sign back in as **userC**
- As **userB**, send **userC** a message
- Notification should be show

**Other**
- [x] The console warning shown above should not appear when testing a dev build
- [x] Verify that no errors appear in the JS console

### Offline tests

**Confirm users are re-registered for notifications when signing out**
_This only works on staging/prod builds_
- Sign in as **userA** on the test platform
- As **userB**, send **userA** a message
- Notification should be show
- As **userA**, sign out
- Without clearing data or killing the window/app, sign back in as **userC**
- As **userC**, disable your network connection
- As **userB**, send **userC** a message
- No notification should be show
- As **userC**, re-enable your network connection
- **userC** should receive the notification

### QA Steps

_The above tests should be run. This should be simpler now that you have testable staging releases. Let me know if you run into any issues!_

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<summary>Web</summary>

<details>

Screenshot 2023-02-13 at 14 42 03

</details>

<details>
<summary>Mobile Web - Chrome</summary>

N/A

</details>

<details>
<summary>Mobile Web - Safari</summary>

N/A

</details>

<details>
<summary>Desktop</summary>

Screenshot 2023-02-13 at 17 54 53

</details>

<details>
<summary>iOS</summary>

IMG_0107

</details>

<details>
<summary>Android</summary>

Screenshot_20230213_151731

</details>

### CODE REVIEW

File: src/libs/Notification/PushNotification/index.native.js [hunk 1]
```diff
@@ -109,12 +106,6 @@ function register(accountID) {
     // Regardless of the user's opt-in status, we still want to receive silent push notifications.
     Log.info(`[PUSH_NOTIFICATIONS] Subscribing to notifications for account ID ${accountID}`);
     UrbanAirship.setNamedUser(accountID.toString());
-
-    // When the user logged out and then logged in with a different account
-    // while the app is still in background, we must resubscribe to the report
-    // push notification in order to render the report click behaviour correctly
```
Comment (Julesssss) [inline src/libs/Notification/PushNotification/index.native.js:115]: I attempted to simplify this comment [here](https://github.com/Expensify/App/pull/14702/files#diff-571033c4e40d8aa5e2b60377b898d82d804e539b987033e6979e89352a567b8aR45)

File: src/libs/Notification/PushNotification/index.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import NotificationType from './NotificationType';
 
 // Push notifications are only supported on mobile, so we'll just noop here
 export default {
+    init: () => {},
```
Comment (Julesssss) [inline src/libs/Notification/PushNotification/index.js:5]: Now that this is actually called outside of the module, I think it makes sense to export it from the non-native file. But I'm not 100% sure it's necessary

**Comment by Julesssss** [comment]
I'm wasting a lot of time here, going to make this a contributor issue

**Comment by Julesssss** [comment]
Reopening as a contributor pointed out that this PR should work. I rebuild a release build, and it does in fact work.

**Comment by Julesssss** [comment]
Tested all platforms except iOS. Will try a full-release build tomorrow then open it up for review.

**Comment by Julesssss** [comment]
CC @abdulrahuman5196

**Review by 0xmiroslav — APPROVED** [review]
Code changes look good to me and verified Tests step working well on web/desktop/android.
NOTE: #15104 is not merged into this PR so `##` fix in android is not applied here
I am not able to test on iOS dev build since I don't have provisioning profile. But I will confirm once staging build is uploaded to Testflight.

**Review by alex-mechler — APPROVED** [review]

**Comment by Julesssss** [comment]
Verified on staging. Signed out and into another account 👍

**Pull request merged**